### PR TITLE
fix(Autocomplete): remove handler to open menu on focus

### LINKED ISSUE #3030: Autocomplete field is broken
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Other

### How is your app built?

Create React App

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### Please describe your bug.

here is demo https://codesandbox.io/s/affectionate-bird-ssdsi5
I already mentoined this bug but got no attention
If you click autocomplete field - select options pops up (ok behavior)
then you select another window (not just defocus a field or select another tab, but entirely different OS window) - select options dissapear (ok)
then you try to click button right below autocomplete - NOPE. autocomplete select options poped up (who asked them?) and covered the button, thus user click lands on the autocomplete, not the button. This is intercepting user input! Facepalm 

### What's the expected behaviour?

When user clicks on something, nothing should appear right below cursor right before click out of nowhere

### Help us reproduce the bug!

I hope this works https://codesandbox.io/s/affectionate-bird-ssdsi5

### Code Snippet

```typescript
// Put your code below this line.

```

### Additional information and screenshots

screenshot will not help, need video.

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

In a special case(#3030), opening dropdown menu on focus is annoying.

#### Issue #, if available

Fix #3030 

#### Description of how you validated changes

Add new unit tests 

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/primitives/types/autocomplete.ts [hunk 1]
```diff
@@ -143,6 +143,4 @@ export interface UseAutocompleteProps extends Partial<AutocompleteProps> {
   onChange: React.ChangeEventHandler<HTMLInputElement>;
 
   onClick: React.MouseEventHandler<HTMLInputElement>;
-
-  onFocus: React.FocusEventHandler<HTMLInputElement>;
```
Comment (zchenwei) [inline packages/react/src/primitives/types/autocomplete.ts:147]: This is internal API

Comment (calebpollman) [reply packages/react/src/primitives/types/autocomplete.ts:147]: Does that mean this is a breaking change 😢

Comment (zchenwei) [reply packages/react/src/primitives/types/autocomplete.ts:147]: Nope, the interface is only used internally

Comment (calebpollman) [reply packages/react/src/primitives/types/autocomplete.ts:147]: Thanks for adding the edit

**Review by calebpollman — APPROVED** [review]

**Review by ioanabrooks — APPROVED** [review]

**Pull request merged**